docs/etl/specializations/README.md

```diff
@@ -0,0 +1,31 @@
+# ETL specializations
+
+This area documents special classes that inherit from ETL base classes to 
+solve specific generic tasks.
+
+## Cache Loader
+
+The Cache Loader is provided as `atc.cache.CacheLoader`. The cached loader 
+passes only some rows on to the write operation, and not all. This can give 
+optimizations for cases where the write operation is costly, such as writes 
+to eventhub or cosmos. 
+
+The following diagram illustrates the flow of data 
+through the cached loader.
+![Cache loader diagram](./cached_loader.svg)
+
+The foundation of the cached loader is a cache table that contains primary key 
+columns and a hash for each row. 
+The cached loader supports the following use-cases:
+- write rows that are 
+  - new wrt. the cache table
+  - changed wrt. the cache table
+- delete rows that are in the cache table but not in the data.
```

Comment:
So the data from the dataplatform (databricks) always makes a Full load when using this class? As I understand it, if you only selected a subset of a databricks gold table, and uses the cache loader, all data (except the subset data) will be deleted - right?

Reply:
Only if you implement the delete functionality, For this, let the `delete_operation` return a dataframe. If `delete_operation` returns None, no deletes will be carried out and no deletes will be registered.